app/models/application_question_set.rb

```diff
@@ -18,12 +18,14 @@ def question_by_attribute(attribute)
   end
 
   def questions
-    form_structure.map do |section|
-      section['questions'].map { |question| ApplicationQuestion.new(question.merge(section: section['title'])) }
-    end.flatten
+    form_structure.map do |_section, section_data|
+      next unless section_data.is_a?(Hash)
+
+      section_data['questions'].map { |question| ApplicationQuestion.new(question.merge(section: section_data['title'])) }
+    end.compact.flatten
   end
 
   def no_of_qs
-    form_structure.map { |section| section[:questions] }.flatten.count
+    form_structure.select { |section| section['questions'] }.flatten.count
```

Comment:
why the change from symbols to strings?

Reply:
After being saved as a jsonb in database section[:questions] is not accessible anymore.

![Screenshot from 2024-08-06 20-32-08](https://github.com/user-attachments/assets/ef2c6483-1852-4673-8e25-33482fae28ba)